internetarchive/cli/ia_upload.py

```diff
@@ -160,19 +163,22 @@ def main(argv, session):
             sys.exit(1)
 
     # Status check.
+    if args['<identifier>']:
+        item = session.get_item(args['<identifier>'])
     if args['--status-check']:
-        if session.s3_is_overloaded():
+        if session.s3_is_overloaded(identifier=args['<identifier>'], access_key=session.access_key):
             print(f'warning: {args["<identifier>"]} is over limit, and not accepting requests. '
                   'Expect 503 SlowDown errors.',
                   file=sys.stderr)
             sys.exit(1)
+        elif item.item_size >= MAX_ITEM_SIZE:
```

Comment:
```suggestion
        elif item.item_size > MAX_ITEM_SIZE:
```

Reply:
This would require some testing whether IA's servers still accept any upload (including an empty file) if the item is exactly 1 TiB. Might be tricky though since I think the metadata files, which get modified after every upload, also count towards the item size.